assets/javascripts/discourse/controllers/preferences-github.js

```diff
@@ -14,10 +13,9 @@ export default class PreferencesGithubController extends Controller {
 
   @action
   login() {
-    let redirectURL = getAbsoluteURL(
-      `/github-verification?user_id=${this.model.id}`
-    );
-    window.location = `https://github.com/login/oauth/authorize?client_id=${this.siteSettings.discourse_github_verification_client_id}&redirect_uri=${redirectURL}`;
+    return ajax(`/github-verification/auth_url`).then((response) => {
+      window.location = response.url;
+    });
```

Comment:
Could make this faster and simpler by doing 

```
window.location = 'getAbsoluteURL(`/github-verification/auth-url`)
```
and then doing a 302 redirect on the server

Reply:
I like that